diff --git a/eth/handler.go b/eth/handler.go
index 1ce792d35..5ca773b8a 100644
--- a/eth/handler.go
+++ b/eth/handler.go
@@ -650,7 +650,7 @@ func (h *handler) BroadcastTransactions(txs types.Transactions) {
 		case tx.Size() > txMaxBroadcastSize:
 			largeTxs++
 		default:
-			numDirect = int(math.Sqrt(float64(len(peers))))
+			numDirect = 0
 		}
 		// Send the tx unconditionally to a subset of our peers
 		for _, peer := range peers[:numDirect] {
