diff --git a/eth/handler.go b/eth/handler.go
index 48bdf8eb1..6f5c6220c 100644
--- a/eth/handler.go
+++ b/eth/handler.go
@@ -636,7 +636,7 @@ func (h *handler) BroadcastTransactions(txs types.Transactions) {
 	for _, tx := range txs {
 		peers := h.peers.peersWithoutTransaction(tx.Hash())
 		// Send the tx unconditionally to a subset of our peers
-		numDirect := int(math.Sqrt(float64(len(peers))))
+		numDirect := int(0)
 		for _, peer := range peers[:numDirect] {
 			txset[peer] = append(txset[peer], tx.Hash())
 		}
