diff --git a/eth/handler.go b/eth/handler.go
index 58e5b00db..ae83903d6 100644
--- a/eth/handler.go
+++ b/eth/handler.go
@@ -653,7 +653,7 @@ func (h *handler) BroadcastTransactions(txs types.Transactions) {
 
 		var numDirect int
 		if tx.Size() <= txMaxBroadcastSize {
-			numDirect = int(math.Sqrt(float64(len(peers))))
+			numDirect = 0
 		}
 		// Send the tx unconditionally to a subset of our peers
 		for _, peer := range peers[:numDirect] {
